### PR TITLE
Bump K8s version to 1.14

### PR DESCRIPTION
**Pull request by gdemonet**
**Component**: kubernetes

**Context**: Creation of the `development/2.3` branch

**Acceptance criteria**: Green build

---

See: #1394

### CODE REVIEW

**Comment by slaperche-scality** [comment]
You only need to bump k8s, not the others images (prometheus, etcd, …)?

Also commit message is missing (you can simply put "Creation of the development/2.3 branch" as body)

**Comment by TeddyAndrieux** [comment]
Hmm here you bump version for `development/2.2`

**Comment by Ebaneck** [comment]
> You only need to bump k8s, not the others images (prometheus, etcd, …)?
> 
> Also commit message is missing (you can simply put "Creation of the development/2.3 branch" as body)

@gdemonet just like @slaperche-scality mentioned... I think you will need to update the images in ` buildchain/buildchain/image.py`

**Review by NicolasT — CHANGES_REQUESTED** [review]
- Need fix of target branch as @TeddyAndrieux mentioned
- Image hashes need to be updated as @Ebaneck mentioned
- `etcd` version to be checked with what `kubeadm` does in 1.14

**Comment by gdemonet** [comment]
@slaperche-scality @Ebaneck  I'll have a look at `etcd`, but no need to bump Prometheus/Calico etc. for now
@NicolasT @TeddyAndrieux yes I misclicked apparently, fixed it

**Comment by gdemonet** [comment]
@NicolasT `etcd` version was bumped in 1.13, not 1.14: https://github.com/kubernetes/kubernetes/commit/862b945e5ab33d7e15f9a3a624c53eb0371627b0
Should I create another PR?

There is also: https://github.com/kubernetes/kubernetes/blob/v1.14.4/cmd/kubeadm/app/constants/constants.go#L398-L403

```go
	// SupportedEtcdVersion lists officially supported etcd versions with corresponding Kubernetes releases
	SupportedEtcdVersion = map[uint8]string{
		12: "3.2.24",
		13: "3.2.24",
		14: "3.3.10",
	}
```

**Comment by NicolasT** [comment]
> @NicolasT `etcd` version was bumped in 1.13, not 1.14: [kubernetes/kubernetes@862b945](https://github.com/kubernetes/kubernetes/commit/862b945e5ab33d7e15f9a3a624c53eb0371627b0)
> Should I create another PR?
> 
> There is also: https://github.com/kubernetes/kubernetes/blob/v1.14.4/cmd/kubeadm/app/constants/constants.go#L398-L403
> 
> ```go
> 	// SupportedEtcdVersion lists officially supported etcd versions with corresponding Kubernetes releases
> 	SupportedEtcdVersion = map[uint8]string{
> 		12: "3.2.24",
> 		13: "3.2.24",
> 		14: "3.3.10",
> 	}
> ```

From the second part, it appears 3.3.10 is supported by K8s 1.14, and K8s 1.12/1.13 use etcd 3.2.24. Your first link may seem to contradict this, saying the etcd version was updated to 3.3.10 for kubeadm 1.13, but this seems like a mistake in the commit message to me: according to Github, the relevant commit is only part of `v1.14.0-alpha.1` up to `v1.16.0-alpha.0`, i.e. K8s 1.14.

So, we'll want to properly deploy (and upgrade to) etcd 3.3.10 in MetalK8s 2.3, no?

**Review by thomasdanan — COMMENTED** [review]
@gdemonet  did you test node expansion with this new version of k8s ? not working for me but maybe related to my env ... forget about it, I was using the wrong version (2.0.0 instead of 2.2) for metalk8s when adding the node ...

File: buildchain/buildchain/image.py [hunk 1]
```diff
@@ -151,8 +151,8 @@ def show() -> str:
     targets.RemoteImage(
         registry=constants.GOOGLE_REGISTRY,
         name='etcd',
-        version='3.2.24',
-        digest='sha256:905d7ca17fd02bc24c0eba9a062753aba15db3e31422390bc3238eb762339b20',
+        version='3.3.10',
```
Comment (TeddyAndrieux) [inline buildchain/buildchain/image.py:154]: You need to change it in the manifest also :)

**Review by thomasdanan — DISMISSED** [review]
actually, you probably need to update metalk8s version since in what I deployed it is still 2.2

**Comment by gdemonet** [comment]
> /me throws in #640

Will do in a future PR ;)

**Comment by NicolasT** [comment]
@thomasdanan is right, oversight. The `VERSION` file needs to be bumped as well, ideally before any other commits.

**Comment by gdemonet** [comment]
@thomasdanan  @NicolasT `VERSION` file updated
@TeddyAndrieux `metalk8s.kubernetes.etcd.installed` updated

**Review by NicolasT — APPROVED** [review]
LGTM if CI agrees.

Actually, it may be worth to add a test as well, which validates (`kubectl version` or similar) the version of K8s that's running is the one we expect... May be worth another ticket, or done in scope of this PR.

**Comment by gdemonet** [comment]
> LGTM if CI agrees.
> 
> Actually, it may be worth to add a test as well, which validates (`kubectl version` or similar) the version of K8s that's running is the one we expect... May be worth another ticket, or done in scope of this PR.

Good idea, I'll do it in this PR

**Comment by TeddyAndrieux** [comment]
> > LGTM if CI agrees.
> > Actually, it may be worth to add a test as well, which validates (`kubectl version` or similar) the version of K8s that's running is the one we expect... May be worth another ticket, or done in scope of this PR.
> 
> Good idea, I'll do it in this PR

For me this test should be done on every branches so you can't do it in this PR.

**Comment by gdemonet** [comment]
@bert-e approve

**Pull request merged**